### PR TITLE
fix otlp recordable_test

### LINKED ISSUE #472: exporters/otlp/recordable_test build fails with grpc
**Describe your environment**
```
opentelemetry-cpp master  c8054c5c46bf81fa57d4d89497fb02b7a863593d
CentOS 8 x64 Stream
cmake = 3.18.4
gcc = 8.4.1
grpc = 1.34  (external lib)
protobuf = 3.14 (external lib)
```
**Steps to reproduce**
```
[user@vm-centos8 build]$ cmake -DBUILD_SHARED_LIBS=ON -DCMAKE_BUILD_TYPE=Release -DWITH_OTLP=ON -DWITH_PROMETHEUS=ON -DWITH_PROTOBUF=ON -DBUILD_TESTING=ON -DWITH_EXAMPLES=ON ..
-- The C compiler identification is GNU 8.4.1
-- The CXX compiler identification is GNU 8.4.1
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Failed
-- Looking for pthread_create in pthreads
-- Looking for pthread_create in pthreads - not found
-- Looking for pthread_create in pthread
-- Looking for pthread_create in pthread - found
-- Found Threads: TRUE  
-- Found Protobuf: /usr/lib64/libprotobuf.so;-lpthread (found version "3.14.0") 
-- Found ZLIB: /usr/lib64/libz.so (found version "1.2.11") 
-- Found OpenSSL: /usr/lib64/libcrypto.so (found version "1.1.1g")  
-- Found c-ares: /usr/lib64/cmake/c-ares/c-ares-config.cmake (found version "1.15.0") 
-- Found RE2 via CMake.
PROTOBUF_PROTOC_EXECUTABLE=/usr/bin/protoc
-- Found GTest: /usr/lib64/libgtest.so  
GTEST_INCLUDE_DIRS   = /usr/include
GTEST_BOTH_LIBRARIES = /usr/lib64/libgtest.so;/usr/lib64/libgtest_main.so
-- Found CURL: /usr/lib64/libcurl.so (found version "7.61.1")  
-- Configuring done
-- Generating done
-- Build files have been written to: /home/user/git/opentelemetry-cpp/build
make
...
...

libopentelemetry_exporter_otprotocol.so: undefined reference to `grpc::ClientContext::~ClientContext()'
libopentelemetry_exporter_otprotocol.so: undefined reference to `grpc::ClientContext::ClientContext()'
libopentelemetry_exporter_otprotocol.so: undefined reference to `grpc::CreateChannel(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::shared_ptr<grpc::ChannelCredentials> const&)'
libopentelemetry_exporter_otprotocol.so: undefined reference to `grpc::g_core_codegen_interface'
libopentelemetry_exporter_otprotocol.so: undefined reference to `grpc::g_glip'
libopentelemetry_exporter_otprotocol.so: undefined reference to `grpc::InsecureChannelCredentials()'
collect2: error: ld returned 1 exit status
make[2]: *** [exporters/otlp/CMakeFiles/recordable_test.dir/build.make:107: exporters/otlp/recordable_test] Error 1
make[1]: *** [CMakeFiles/Makefile2:3593: exporters/otlp/CMakeFiles/recordable_test.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
```

### PR DESCRIPTION
**Pull request by lalitb**
Fixes #472 
And adds workflow to run otlp exporter tests in `ubuntu-latest`

### CODE REVIEW

File: ci/setup_grpc.sh [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+pushd $BUILD_DIR
+git clone --recurse-submodules -b v1.34.0 https://github.com/grpc/grpc
+cd grpc
+mkdir -p cmake/build
+pushd cmake/build
+cmake -DgRPC_INSTALL=ON \
+    -DgRPC_BUILD_TESTS=OFF \
+    -DCMAKE_INSTALL_PREFIX=$INSTALL_DIR \
+    ../..
+make -j 6
```
Comment (ThomsonTan) [inline ci/setup_grpc.sh:21]: Is this `-j 6` necessary to be specified manually?

File: ci/setup_grpc.sh [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+make install
+popd
+popd
+export PATH=${INSTALL_DIR}/bin:$PATH  # nsure to use the installed grpc
```
Comment (ThomsonTan) [inline ci/setup_grpc.sh:25]:
```suggestion
export PATH=${INSTALL_DIR}/bin:$PATH  # ensure to use the installed grpc
```

And `usr/local/bin` should already be part of the PATH?

**Comment by ThomsonTan** [comment]
gRPC build/installation may take too much time which slows down our CI process, probably we could customize the our container with gRPC preinstalled. @maxgolov

File: ci/setup_grpc.sh (hunk 1 above)
Comment (lalitb) [reply ci/setup_grpc.sh:21]: This seems to be optimal value as I tested. `make -j` creates lots of jobs simultaneously and the process crashes in end. `make` is too slow.

File: ci/setup_grpc.sh (hunk 2 above)
Comment (lalitb) [reply ci/setup_grpc.sh:25]: Yes, just made it ahead of `/usr/bin` to use the currently installed once, and not default ( it already installed )

**Comment by lalitb** [comment]
> gRPC build/installation may take too much time which slows down our CI process, probably we could customize the our container with gRPC preinstalled. @maxgolov
Yes, it definitely takes time to install.  If there are github vm's preinstalled with grpc, or else once we move to container based pre-built images, CI would be faster.

File: ci/setup_grpc.sh (hunk 1 above)
Comment (pyohannes) [reply ci/setup_grpc.sh:21]: Usually this works well:
```
make -j $(nproc)
```
`nproc` returns the number of cores a machine has (or a docker container has assigned to it).

**Review by ThomsonTan — APPROVED** [review]
LGTM

File: ci/setup_grpc.sh (hunk 1 above)
Comment (lalitb) [reply ci/setup_grpc.sh]: thanks, changed to `nproc` :)

**Review by jsuereth — APPROVED** [review]

**Pull request merged**